File: backend/models/Food.js
const mongoose = require('mongoose');

const FoodSchema = new mongoose.Schema({
  name: { type: String, required: true },
  price: { type: Number, required: true },
  quantity: { type: Number, default: 0 },
  available: { type: Boolean, default: true },
  category: { type: String, enum: ['restaurant','bakery'], required: true }
}, { timestamps: true });

module.exports = mongoose.model('Food', FoodSchema);

File: backend/src/main/java/com/bojunka/backend/model/Food.java
package com.bojunka.backend.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "foods")
public class Food {
    @Id
    private String id;
    private String name;
    private Double price;
    private String category; // 'restaurant' or 'bakery'
    private Integer quantity;
}
